clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2567,8 +2564,11 @@ private void maybeAddConsumerGroup(ListGroupsResponse.Group group) {
                         void handleResponse(AbstractResponse abstractResponse) {
                             final ListGroupsResponse response = (ListGroupsResponse) abstractResponse;
                             synchronized (results) {
-                                if (response.error() != Errors.NONE) {
-                                    results.addError(response.error().exception(), node);
+                                Errors error = response.error();
```

Comment:
This is very similar to `handleRetriableError`, perhaps there is an opportunity to use a common code.